cadc-util/src/main/java/ca/nrc/cadc/net/DigestUtil.java

```diff
@@ -0,0 +1,225 @@
+ *
+ *  : 5 $
+ *
+ ************************************************************************
+ */
+
+package ca.nrc.cadc.net;
+
+import ca.nrc.cadc.util.StringUtil;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import org.apache.log4j.Logger;
+
+/**
+ * Utility class for the Digest HTTP header.
+ */
+public class DigestUtil {
+    private static Logger log = Logger.getLogger(DigestUtil.class);
+
+    /**
+     * Supported algorithm's in DigestUtil header.
+     */
+    public enum Algorithm {
+        MD5("md5"),
+        SHA1("sha-1"),
+        SHA256("sha-256"),
+        SHA384("sha-384"),
+        SHA512("sha-512");
+
+        public final String value;
+
+        private Algorithm(String value) {
+            this.value = value;
+        }
+
+        public static Algorithm fromString(String value) {
+            for (Algorithm algorithm : Algorithm.values()) {
+                if (algorithm.value.equalsIgnoreCase(value)) {
+                    return algorithm;
+                }
+            }
+            throw new IllegalArgumentException(String.format("Algorithm not found for value %s", value));
+        }
+    }
+
+    /**
+     * Parse the given digest string and return a URI of the form: {algorithm}:{checksum in hex}
+     * @param digest value to parse
+     * @return digest URI
+     */
+    public static URI getURI(String digest) {
+        if (!StringUtil.hasLength(digest)) {
+            return null;
+        }
+
+        int index = digest.indexOf('=');
+        if (index == -1) {
+            throw new IllegalArgumentException(String.format("Expected digest format <algorithm>=<checksum value> "
+                                                                 + "found %s", digest));
+        }
+        String scheme = digest.substring(0, index).trim();
+        String checksumValue = digest.substring(index + 1).trim();
+
+        if (scheme.length() == 0) {
+            throw new IllegalArgumentException(String.format("Unable to parse algorithm, expected format "
+                                                                 + "<algorithm>=<checksum value> found %s",
+                                                             digest));
+        }
+        if (checksumValue.length() == 0) {
+            throw new IllegalArgumentException(String.format("Unable to parse checksum, expected format "
+                                                                 + "<algorithm>=<checksum value> found %s",
+                                                             digest));
+        }
+
+        Algorithm algorithm;
+        try {
+            algorithm = Algorithm.fromString(scheme);
+        } catch (IllegalArgumentException e) {
+            throw new IllegalArgumentException(String.format("Unsupported algorithm %s", scheme));
+        }
+
+        // For a MD5 checksum, determine if the checksum value is hex or base64
```

Comment:
Noooooo!!!!! Don't re=introduce the old bug here! Spec says md5 values are base64 encoded so only support that.